code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -15,6 +15,8 @@
 	var/obj/item/storage/pill_bottle/bottle = null
 	var/mode = 1
 	var/condi = FALSE
+	var/advanced = FALSE
```

Comment:
These should be combined into just one bitflag, i.e. `var/condition`

Reply:
Actually, not a bitflag, since there's no defined functionality for something being both primitive and advanced. It should be an enum:
```
#define CM_COND_PRIMITIVE 1
#define CM_COND_ADVANCED 2
```
and equality should be checked, rather than using bit math.

Reply:
`condi`, `advanced`, and `primitive` could probably all be rolled into one, actually

Reply:
Right, I can't wrap my head around this atleast with JS. Most of the options I tried usually end up in TGUI bluescreens.